src/Pages/ProductDetails/ProductDetailsHeader/ProductDetailsHeader.component.js

```diff
@@ -0,0 +1,214 @@
+    try {
+      const { productId, totalAmount } = this.state;
+      const response = await fetch(API, {
+        method: "POST",
+        headers: {
+          Authorization:
+            "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiaGFydW0ifQ.nMUcgev8vz4rbQY-3z2F0tFFSKQjBMgwCVWOOTm91Qw",
+        },
+        body: JSON.stringify({
+          productId,
+          quantity: totalAmount,
+        }),
+      });
+      if (response.state === 200) {
+        throw new Error("cannot fetch the data");
+      }
+      const result = await response.json();
+      console.log(result);
+    } catch (err) {
+      console.log("!! error alert !!");
+    }
+  };
+
+  componentDidMount = () => {
+    const { productDetail } = this.props;
+    this.setState({
+      rootPrice: this.getRealPrice(productDetail),
+      totalPrice: this.getRealPrice(productDetail),
+      totalAmount: 1,
+      productId: productDetail.id,
+    });
+  };
+
+  render() {
+    const {
+      name,
+      size,
+      content,
+      imageUrl,
+      originalPrice,
+      discountPrice,
+      discountContent,
+      salesUnit,
+      otherInformation,
+    } = this.props.productDetail;
+    const {
+      totalPrice,
+      totalAmount,
+      isMinusAmount,
+      isWidthBiggerThanHeight,
+    } = this.state;
+    return (
+      <div className="ProductDetailsHeader">
+        <figure className="product-image">
+          <img
+            src={imageUrl}
+            className={isWidthBiggerThanHeight ? "full-height" : "full-width"}
+            onLoad={this.resizeImage}
+            alt=""
+          />
+        </figure>
+        <div className="orderinfo">
+          <div className="order-header">
+            <div>
+              <span>{name}</span>
+              <span>{content}</span>
+            </div>
+            <button>
+              <i className="far fa-share-alt"></i>
+            </button>
+          </div>
+          <div className="order-price">
+            <div>회원할인가</div>
+            <div className="sale-price">
+              <span>
+                {discountPrice
+                  ? this.getNumberWithCommas(discountPrice)
+                  : this.getNumberWithCommas(originalPrice)}
+              </span>
+              <span>원</span>
+              <span>{discountContent}</span>
+            </div>
+            <div
+              className={`original-price ${!discountPrice && "display-none"}`}
+            >
+              <span>{originalPrice}</span>
+              <span>원</span>
+              <i className="far fa-question-circle"></i>
+            </div>
+            <div>로그인 후, 회원할인가와 적립혜택이 제공됩니다.</div>
+          </div>
+          <div className="order-info">
+            <dl>
+              <dt>판매단위</dt>
+              <dd>{salesUnit}</dd>
+            </dl>
+            <dl>
+              <dt>중량/용량</dt>
+              <dd>{size}</dd>
+            </dl>
+            {Object.entries(otherInformation).map((info, idx) => {
```

Comment:
```suggestion
            {Object.entries(otherInformation).map(([ key, value ], idx) => {
```
- info라는 변수명 보다 구체적인 이름이 있을 것 같습니다.
- 애초에 info를 곧바로 구조분해 할당하면서 0, 1번 인덱스에 들어올 값의 이름을 정해줄 수 있습니다. key value라고 적어두긴 했지만 각각의 아이템에 대한 적절한 이름을 지어주세요.

Reply:
```jsx
           {otherInformation.map((info, idx) => {
              return (
                info.description && (
                  <dl key={idx}>
                    <dt>{info.category}</dt>
                    <dd>{info.description}</dd>
                  </dl>
                )
              );
            })}
```
Backend 랑 얘기해서 다음과 같이 정리했습니다! 🙌🏻 🙌🏻 🙌🏻 🙌🏻 🙌🏻